src/controllers/sync-controller.ts

```diff
@@ -318,14 +326,55 @@ export class SyncController implements ReactiveController {
     }
   }
 
-  private async checkSyncStatus() {
+  /**
+   * Check if a sync was interrupted and resume it if necessary
+   * Handles all sync phases, not just assets and read status
+   */
+  async #checkAndResumeSync() {
     try {
-      // Check if there are bookmarks that still need asset sync
-      const bookmarksNeedingAssetSync = await DatabaseService.getBookmarksNeedingAssetSync();
-      const bookmarksNeedingReadSync = await DatabaseService.getBookmarksNeedingReadSync();
-
+      // Check multiple conditions that indicate an interrupted sync:
+      // 1. Last sync timestamp vs current time (incomplete incremental sync)
+      // 2. Bookmarks needing asset sync (phase 3)
+      // 3. Bookmarks needing read sync (phase 4)
+      // 4. Sync state in database indicating interruption
+
+      const [lastSyncTimestamp, bookmarksNeedingAssetSync, bookmarksNeedingReadSync] = await Promise.all([
+        DatabaseService.getLastSyncTimestamp(),
+        DatabaseService.getBookmarksNeedingAssetSync(),
+        DatabaseService.getBookmarksNeedingReadSync()
+      ]);
+
+      // Determine if sync was interrupted based on various indicators
+      let needsResume = false;
+      let resumeReason = '';
+
+      // Check for incomplete sync phases 3-4
       if (bookmarksNeedingAssetSync.length > 0 || bookmarksNeedingReadSync.length > 0) {
-        DebugService.logInfo('sync', `Resuming sync: ${bookmarksNeedingAssetSync.length} assets, ${bookmarksNeedingReadSync.length} read status`);
+        needsResume = true;
+        resumeReason = `${bookmarksNeedingAssetSync.length} assets, ${bookmarksNeedingReadSync.length} read status`;
+      }
+
+      // Check for very recent sync timestamp that might indicate interruption during phases 1-2
+      // If last sync was within the last 5 minutes but we still have pending items, likely interrupted
+      if (lastSyncTimestamp) {
+        const timeSinceLastSync = Date.now() - new Date(lastSyncTimestamp).getTime();
+        const fiveMinutesMs = 5 * 60 * 1000;
+
+        if (timeSinceLastSync < fiveMinutesMs && !needsResume) {
+          // Recent sync but check if there are any sync state indicators
+          const [unarchivedOffset, archivedOffset] = await Promise.all([
+            DatabaseService.getUnarchivedOffset(),
+            DatabaseService.getArchivedOffset()
+          ]);
+          if (unarchivedOffset > 0 || archivedOffset > 0) {
+            needsResume = true;
+            resumeReason = `Incomplete bookmark sync (offsets: unarchived=${unarchivedOffset}, archived=${archivedOffset})`;
+          }
+        }
```

Comment:
Why do we only resume if it's been less than 5 minutes? Doesn't the presence of non-zero offsets indicate that the sync was interrupted, and hence should be resumed regardless of how long it's been? Or is there a situation where we would not want to resume if it's been more than 5 minutes but the offsets are non-zero?